### PR TITLE
feat: Implement evm_increaseTime and evm_setNextBlockTimestamp

### LINKED ISSUE #66: evm_api: Implement `evm_increaseTime` API endpoint
### **Description:**
The `evm_increaseTime` endpoint needs to be implemented. The purpose of this endpoint is to jump forward in time by the given amount of time, in seconds. [Ganache docs for reference](https://ganache.dev/#evm_increaseTime).

```rust
// Interface in the `trait`
#[rpc(name = "evm_increaseTime")]
fn increase_time(
    &self,
    time_delta: U64,
) -> BoxFuture<Result<U64>>;

// Actual code in the `impl`
fn increase_time(
    &self,
    time_delta: U64,
) -> BoxFuture<Result<U64>> {
    // TODO: Solution here
}
```

#### Expected response
Command returns proper response:
```
curl -X POST http://localhost:8011 \
     -H "Content-Type: application/json" \
     --data '{
       "jsonrpc": "2.0",
       "id": 1,
       "method": "evm_increaseTime",
       "params": ["3600"]
     }'
```

Response:
```
{"jsonrpc":"2.0", "result":"3600", "id":1}
```

### **Requirements:**

1. **Implement the API:**
    - The endpoint should return a populated `U64` of the total time adjustment, in seconds.
    - Ensure the method executes the call reliably and efficiently.
    - Ensure errors are properly returned.

2. **Unit Tests:**
    - Write unit tests to verify the functionality of the `increase_time` method.
    - Ensure coverage for multiple edge-case scenarios.

3. **Documentation:**
    - Add inline Rust documentation (`///` comments) detailing the method's purpose, return values, and any potential edge cases or error scenarios.
    - Update `SUPPORTED_APIS.md` with updated documentation and examples.
    - (Optional) Add request to `test_endpoints.http` for easier manual verification.

### **Task Checklist:**
- [ ] Implement the `increase_time` method.
- [ ] Write unit tests for the `increase_time` method.
- [ ] Add inline documentation for the `evm_increaseTime` endpoint.
- [ ] Update documentation for the `evm_increaseTime` endpoint in `SUPPORTED_APIS.md`.

### **Additional Notes:**
If you're an external contributor looking to take on this issue, please comment below to let us know you're working on it. If you need further clarification on the requirements or need assistance with anything related to the implementation, feel free to ask!

### PR DESCRIPTION
**Pull request by nbaztec**
# What :computer: 

Adds the following EVM rpc methods to the test node:

* `evm_increaseTime`
* `evm_setNextBlockTimestamp`
* * `evm_setTime`

# Why :hand:
* Allows users to manipulate the node timestamp as per there specific preferences, which can be highly beneficial during testing.

# Evidence :camera:
Passing unit tests:
image

# Notes :memo:

Fixes #66 & #68

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1780,6 +1780,7 @@ dependencies = [
  "jsonrpc-http-server",
  "lazy_static",
  "once_cell",
+ "openssl-sys",
```
Comment (MexicanAce) [inline Cargo.lock:1783]: Hmm, perhaps this happened in the middle of the merge from `main`. Can you revert the changes to `Cargo.lock`? Everything else looks good 👍

Comment (MexicanAce) [reply Cargo.lock:1783]: Oh wait, I'm seeing the same thing locally... I think we missed something in the latest PR to `main`

Comment (MexicanAce) [reply Cargo.lock:1783]: K, leave as is. This was a mistake with previous PR.

**Comment by nbaztec** [comment]
@MexicanAce I changed the return types to native types `u64` to allow for rpc calls to be numeric instead of the hex encoding that `U64` requires.

File: SUPPORTED_APIS.md [hunk 2]
```diff
@@ -86,7 +86,7 @@ The `status` options are:
 | `EVM` | `evm_setBlockGasLimit` | `NOT IMPLEMENTED` | Sets the Block Gas Limit of the network |
 | `EVM` | `evm_setIntervalMining` | `NOT IMPLEMENTED` | Enables (with a numeric argument greater than 0) or disables (with a numeric argument equal to 0), the automatic mining of blocks at a regular interval of milliseconds, each of which will include all pending transactions |
 | `EVM` | `evm_setNextBlockTimestamp` | `NOT IMPLEMENTED`<br />[GitHub Issue #68](https://github.com/matter-labs/era-test-node/issues/68) | Works like `evm_increaseTime`, but takes the exact timestamp that you want in the next block, and increases the time accordingly |
-| `EVM` | `evm_setTime` | `NOT IMPLEMENTED` | Sets the internal clock time to the given timestamp |
```
Comment (MexicanAce) [inline SUPPORTED_APIS.md:89]: I think there's actually a need for both here. Ganache only supports [evm_setTime](https://ganache.dev/#evm_setTime), while Hardhat only support [evm_setNextBlockTimestamp](https://hardhat.org/hardhat-network/docs/reference#evm_setnextblocktimestamp) ... but I _think_ they both do the same thing. Thoughts?

Comment (MexicanAce) [reply SUPPORTED_APIS.md:89]: We can either implement `evm_setTime` in this PR, or leave the line here in this table.

Also, it looks like this change leaves 2 rows/entries for `evm_setNextBlockTimestamp`

Comment (nbaztec) [reply SUPPORTED_APIS.md:89]: > I think there's actually a need for both here. Ganache only supports [evm_setTime](https://ganache.dev/#evm_setTime), while Hardhat only support [evm_setNextBlockTimestamp](https://hardhat.org/hardhat-network/docs/reference#evm_setnextblocktimestamp) ... but I _think_ they both do the same thing. Thoughts?

Yeah they do exactly the same thing with the following differences:
* `setNextBlockTimestamp` allows only setting a timestamp forward in future, while `setTime` supports setting it backwards.
* `setNextBlockTimestamp` returns the newly set timestamp in response, whereas `setTime` returns the difference between the current and the new timestamp.

AFAIK `setTime` can be seen as a superset of `setNextBlockTimestamp` and from most examples out there no one seems to be using the return value to do anything meaningful.

I'm fine with supporting both, though it must be explicitly mentioned that these two achieve the same result and are mostly arising out of api differences between hardhat and ganache - though it does tend to make our API interface a bit messy.

**Comment by dutterbutter** [comment]
@nbaztec thanks for the contribution! Can you address the conflicts on this PR please.

**Comment by dutterbutter** [comment]
@nbaztec the Headers timeout issue in the build fail is known and should be resolved this week in an upcoming release.

**Review by MexicanAce — APPROVED** [review]
LGTM
